modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/TransferableObjectProcessor.java

```diff
@@ -228,11 +231,22 @@ private void validateMessages(List<MessageClass> messages) {
     /**
      * Extracts and validates the declared message group types marked with the {@link MessageGroup} annotation.
      */
-    private static MessageGroupWrapper getMessageGroup(RoundEnvironment roundEnv) {
+    private MessageGroupWrapper getMessageGroup(RoundEnvironment roundEnv) {
         Set<? extends Element> messageGroupSet = roundEnv.getElementsAnnotatedWith(MessageGroup.class);
 
         if (messageGroupSet.isEmpty()) {
-            throw new ProcessingException("No message groups (classes annotated with @MessageGroup) found");
+            Elements elements = processingEnv.getElementUtils();
+
+            Set<String> packageNames = roundEnv.getRootElements().stream()
+                    .map(elements::getPackageOf)
+                    .map(PackageElement::getQualifiedName)
+                    .map(Name::toString)
+                    .collect(Collectors.toSet());
+
+            throw new ProcessingException(
+                    "No message groups (classes annotated with @MessageGroup) found while processing messages from the following packages: "
```

Comment:
Maybe use: "@" + MessageGroup.class.getSimpleName()
In case the name of the annotation changes.